clustermesh/clustermesh.go

```diff
@@ -2158,3 +2250,59 @@ func updateClustermeshConfig(
 
 	return newValues, nil
 }
+
+func removeFromClustermeshConfig(values map[string]interface{}, aiRemote *accessInformation) (map[string]any, error) {
+	// get current clusters config slice, if it exists
+	c, found, err := unstructured.NestedFieldCopy(values, "clustermesh", "config", "clusters")
+	if err != nil {
+		return nil, fmt.Errorf("existing clustermesh.config is invalid")
+	}
+	if found && c == nil {
+		return nil, fmt.Errorf("active cluster is not connected to any other clusters")
+	}
+	if !found {
+		c = []any{}
+	}
+
+	cs, ok := c.([]any)
+	if !ok {
+		return nil, fmt.Errorf("existing clustermesh.config.clusters array is invalid")
+	}
+	oldClusters := make([]map[string]any, 0, len(cs))
+	for _, m := range cs {
+		if cluster, ok := m.(map[string]any); ok {
+			oldClusters = append(oldClusters, cluster)
+			continue
+		}
+		return nil, fmt.Errorf("existing clustermesh.config.clusters map is invalid")
+	}
+
+	clusters, victim := map[string]map[string]any{}, false
+	for _, c := range oldClusters {
+		if name, ok := c["name"].(string); ok {
+			if name == aiRemote.ClusterName {
+				victim = true
+				continue
+			}
+			clusters[name] = c
+		}
+	}
+	if !victim {
+		return nil, fmt.Errorf("active cluster is not currently connected to cluster %q", aiRemote.ClusterName)
+	}
+
+	outputClusters := make([]map[string]any, 0, len(clusters))
+	for _, v := range clusters {
+		outputClusters = append(outputClusters, v)
+	}
```

Comment:
This logic could be likely simplified to avoid creating multiple intermediate maps. Given that no subsequent modification needs to be performed, a single pass which copies all existing elements (but the victim) to a new slice should be enough.

Reply:
Good point! Thank you Marco!